cli/azd/test/functional/cli_test.go

```diff
@@ -108,6 +108,42 @@ func Test_CLI_Init_CanUseTemplate(t *testing.T) {
 	require.FileExists(t, filepath.Join(dir, "README.md"))
 }
 
+// Test_CLI_Up_CanUseTemplateWithoutExistingProject ensures that you can run `azd up --template <some-template>` in an
+// empty directory and the project will be initialize as expected.
+func Test_CLI_Up_CanUseTemplateWithoutExistingProject(t *testing.T) {
+	// running this test in parallel is ok as it uses a t.TempDir()
+	t.Parallel()
+	ctx, cancel := newTestContext(t)
+	defer cancel()
+
+	dir := tempDirWithDiagnostics(t)
+
+	cli := azdcli.NewCLI(t)
+	cli.WorkingDirectory = dir
+	cli.Env = append(os.Environ(), "AZURE_LOCATION=eastus2")
```

Comment:
Another option would be to create this test in the `actions_test.go`.  We don't really care about the result of the command, only that the is able to be instantiated, right?

Reply:
In the existing tests we always create an `azure.yaml` - that is why this wasn't caught earlier.

Reply:
> Another option would be to create this test in the actions_test.go. We don't really care about the result of the command, only that the is able to be instantiated, right?

Yeah - I think we would be fine just validating you can create the action.  I'll see if I can do something better in `actions_test.go` and remove this.

Reply:
I'm going to keep this end to end test as is now - I think there is some value over doing it in `actions_test.go`, we get better coverage and it is closer to one of our main gestures, so I think I want to keep the test for now.